File: api/app/Models/vehiculo.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class vehiculo extends Model
{
    protected $table = 'vehiculos';
    protected $fillable = ['NSerie','marca','modelo','color','año','img','precio','kilometros','descripcion','id_user'];

    public function user(){
        return $this->HasOne(User::Class,'id','id_user');
    }
    public function marca(){
        return $this->HasOne(Marca::Class,'id','id_marca');
    }
    public function categoria(){
        return $this->HasOne(Category::Class,'id','id_categoria');
    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/models/Vehiculo.kt
package com.example.carbid.models

import java.time.Year

data class Vehiculo(val id: Int, val NSerie: String, val Marca: String,
                    val Modelo: String,val year: Int, val desc: String,
                    val precio: Double, val imagen: String, val ubicacion: String )
